### PR TITLE
add unmarshel func for RoutingTables.Routes

### PR DESCRIPTION
**Pull request by olasaadi99**

### CODE REVIEW

**Review by adisos — CHANGES_REQUESTED** [review]
looks good, few comments:

- [ ] Add the relevant test JSON that demonstrated this issue, into the collector's unmarshal test.

- [ ] Figure out how the unmarshal works for the `Instance` with `NetworkInterfaces` without a wrapper type and dedicated `UnmarshalJSON` function.

File: pkg/ibm/datamodel/test/unmarshal_test.go [hunk 1]
```diff
@@ -16,10 +16,11 @@ import (
 )
 
 func TestUnmarshal(t *testing.T) {
-	unmarshalInputs := [3]string{
+	unmarshalInputs := [4]string{
```
Comment (adisos) [inline pkg/ibm/datamodel/test/unmarshal_test.go:19]:
```suggestion
	unmarshalInputs := []string{
```

**Comment by adisos** [comment]
>     * [x]  Figure out how the unmarshal works for the `Instance` with `NetworkInterfaces` without a wrapper type and dedicated `UnmarshalJSON` function.

Summary of the discussion with @olasaadi99 : For the `RoutingTable` type there was a need to add a wrapper type to `Route`, since the standard `unmarshal` failed to unmarshal `Route` object, because it contained an `interface` type field for `next_hop`.
The custom unmarshal `UnmarshalJSON`  added to the wrapper type - called `vpcv1.UnmarshalRoute`, which could handle the unmarshaling of Route objects including this `next_hop` field.. 
(In the case of `NetworkInterfaces` for `Instance` the standard `unmarshal` worked, because this struct did not contain fields of type interface). 

@kyorav @zivnevo  I think we might want to change the unmarshaling of the collector to always call the dedicated `Unmarshal` functions of the `vpcv1` pkg , so that we are not susceptible to fields of interface types that fail the standard unmarshal function. I think this would be more robust and would eliminate the "wrapper types" work-around used currently. Let me know what you think.

**Comment by kyorav** [comment]
I am not sure how this would work. Our model is different from the vpc1, so we still need to write our own un-marsheling function in certain places. If I remember correctly, as soon as you call the vpc1 function you cannot interrupt it with your own code (without modifying the vpc1 module). Can you sketch what you mean by "always call the dedicated Unmarshal functions of the vpcv1 pkg"?

**Comment by adisos** [comment]
> I am not sure how this would work. Our model is different from the vpc1, so we still need to write our own un-marsheling function in certain places. If I remember correctly, as soon as you call the vpc1 function you cannot interrupt it with your own code (without modifying the vpc1 module). Can you sketch what you mean by "always call the dedicated Unmarshal functions of the vpcv1 pkg"?

I think we can implement `UnmarshalJSON` for every type of our data_model, for example it was already implemented for `EndpointGateway`, where it calls `vpcv1.UnmarshalEndpointGateway` on the field that is imported from `vpc1`. 

Similarly, for our model's `RoutingTable` , we can have `UnmarshalJSON`  that calls both `UnmarshalRoutingTable` and `UnmarshalRoute` for the list of routes, and assigns the unmarshaled objects in their fields of the `RoutingTable`  object. 

The goal is that we don't use the standard `unmarshal` function for `vpc1` fields in our model.

**Comment by adisos** [comment]
Approving this PR, will continue the discussion offline, and open a separate issue if we decide to change the `unmarshal` implementation.

**Review by adisos — APPROVED** [review]

**Pull request merged**